cpp/src/arrow/util/compression.h

```diff
@@ -30,7 +30,18 @@ namespace arrow {
 
 struct Compression {
   /// \brief Compression algorithm
-  enum type { UNCOMPRESSED, SNAPPY, GZIP, BROTLI, ZSTD, LZ4, LZ4_FRAME, LZO, BZ2 };
+  enum type {
```

Comment:
Are those the same enums as exposed by Parquet public API?

If so, this would be a break of ABI compatibility. Also could we start explicitly numbering these, and maybe add a comment, so that the next person doesn't break ABI compatibility without realising it?

Reply:
We don't guarantee ABI stability for Arrow/Parquet C++, so this is not really a problem.
That said, I could change this if people feel strongly about it.

Reply:
It would be much appreciated as it would help with our project:
https://github.com/G-Research/ParquetSharp

Currently, much of the C++ ABI is abstracted away, except for enum values

Reply:
I see. You should be aware that this enum was already changed at some point in 895f22009f945371fb18d00324cc5eb853dda51a .

Reply:
That was before 1.0.0 though.

Reply:
Yes. You're correct. I double checked when I saw this PR and our code is currently wrong. I'll probably add a few static asserts to make sure we catch it in the future. 

However, this would still in turn break our .NET ABI compatibility. The proper fix on our side would be to completely abstract the enum values and translate them both ways. Which is why I'd favour explicit enum values in Arrow, as I think it's relatively cheap and would make our life much easier 😁

Reply:
I can simply move the new enum value at the end.